mantidimaging/gui/windows/spectrum_viewer/presenter.py

```diff
@@ -244,28 +247,29 @@ def handle_notify_roi_moved(self, roi: SpectrumROI) -> None:
         self.update_roi_on_fitting_thumbnail()
 
     def run_spectrum_calculation(self):
-        roi_name = list(self.roi_to_process_queue.keys())[0]
-        roi = self.roi_to_process_queue[roi_name]
-        chunk_size = 100
-        if chunk_size > 0:
-            nanInds = np.argwhere(np.isnan(self.image_nan_mask_dict[roi_name]))
-            chunk_start = int(nanInds[0, 0])
-            if len(nanInds) > chunk_size:
-                chunk_end = int(nanInds[chunk_size, 0])
+        with self.spectrum_calculation_lock:
```

Comment:
I am not sure if this solves the problem. We still access lots of presenters variables from inside the worker thread, which might simultaneously be access/modified in the main thread. I think this approach would also need the lock to be acquired everywhere that things like `self.roi_to_process` are accessed.

Would it work to put input data in the `TaskWorkerThread` kwargs and then return the spectrum chunk and put in to `spectrum_data_dict` from in `thread_cleanup`?

Or would it be possible to put the tasks that need doing in a [queue](https://docs.python.org/3/library/queue.html)?